wifiphisher/common/interfaces.py

```diff
@@ -465,26 +514,91 @@ def get_ap_iface(self, interface_name=None):
+        Check if the interface is managed by NetworkManager
+
+        :param self: A NetworkManager object
+        :type self: NetworkManager
+        :param iface: Interface name
+        :type iface: str
+        :return True if nterface is managed by NetworkManager
+        :rtype bool
+        """
+        bus = dbus.SystemBus()
+        nm_proxy = bus.get_object(NM_APP_PATH, NM_MANAGER_OBJ_PATH)
+        nm = dbus.Interface(nm_proxy, dbus_interface=NM_MANAGER_INTERFACE_PATH)
+        devs = nm.GetDevices()
+        is_managed = False
+        for dev_obj_path in devs:
+            dev_proxy = bus.get_object(NM_APP_PATH, dev_obj_path)
+            dev = dbus.Interface(dev_proxy, dbus_interface=dbus.PROPERTIES_IFACE)
+            if dev.Get(NM_DEV_INTERFACE_PATH, 'Interface') == iface:
+                is_managed = dev.Get(NM_DEV_INTERFACE_PATH, 'Managed')
+        return is_managed
+
     def set_internet_iface(self, iface):
-        if pyw.iswireless(iface):
-            raise Exception
         self.internet_iface = iface
+        if pyw.iswireless(iface):
+            iface_obj = self._interfaces[iface]
+            iface_obj.being_used = True
+            iface_obj.is_internet_iface = True
+            self._interfaces[iface] = iface_obj
 
     def set_ap_iface(self, iface):
+        """
+        Specify the ap interface
+        
+        :param self: A NetworkManager object
+        :param iface: AP interface Name
+        :type self: NetworkManager
+        :type iface: str
+        :return: None
+        """
         self.ap_iface = iface
         iface_obj = self._interfaces[iface]
         iface_obj.being_used = True
         self._interfaces[iface] = iface_obj
 
     def set_jam_iface(self, iface):
+        """
+        Specify the jamming interface
+        
+        :param self: A NetworkManager object
+        :param iface: Deauth interface name
+        :type self: NetworkManager
+        :type iface: str
+        :return: None
+        :rtype: None
+        """
         self.jam_iface = iface
         iface_obj = self._interfaces[iface]
         iface_obj.being_used = True
         self._interfaces[iface] = iface_obj
+   
+    def check_ifaces_uncontrolled_by_nm(self):
+        """
+        Check the interfaces are uncontrolled by NM
+        
+        :param self: A NetworkManager object
+        :type self: NetworkManager
+        :return: None
+        :rtype: None
+        :raises DeauthInterfaceManagedByNMError if deauth interface is in managed state
+        :raises ApInterfaceManagedByNMError if ap interface is in managed state
+        """
+
+        for iface, iface_obj in self._interfaces.iteritems():
+            if iface == self.jam_iface:
+                if self._is_iface_managed_by_nm(iface):
+                    raise DeauthInterfaceManagedByNMError
+            elif iface == self.ap_iface:
+                if self._is_iface_managed_by_nm(iface):
+                    raise ApInterfaceManagedByNMError 
```

Comment:
Maybe we should keep things simple. Instead of having two separate errors, we can have one generic "InterfaceManagedByNetworkManagerError".

Then the code would be:

```python
if iface == self.jam_iface or iface==self.ap_iface:
    if self._is_iface_managed_by_nm(iface):
        raise InterfaceManagedByNetworkManagerError
```